src/generic/upstream-info-icon/index.tsx

```diff
@@ -1,41 +1,138 @@
 /* eslint-disable react/prop-types */
 import { useIntl } from '@edx/frontend-platform/i18n';
-import { Icon } from '@openedx/paragon';
-import { LinkOff, Newsstand } from '@openedx/paragon/icons';
+import {
+  Button, Icon, OverlayTrigger, Tooltip,
+} from '@openedx/paragon';
+import {
+  CallSplit, LinkOff, Newsstand, Sync,
+} from '@openedx/paragon/icons';
 
+import { BoldText } from '@src/utils';
+import { ReactNode } from 'react';
 import messages from './messages';
 
 export interface UpstreamInfoIconProps {
   upstreamInfo?: {
     errorMessage?: string | null;
     upstreamRef?: string | null;
+    upstreamName: string;
+    readyToSync: boolean;
+    downstreamCustomized: string[];
   };
   size?: 'xs' | 'sm' | 'md' | 'lg' | 'inline';
 }
 
-export const UpstreamInfoIcon: React.FC<UpstreamInfoIconProps> = ({ upstreamInfo, size }) => {
+const UpstreamInfoIconContent = ({
+  upstreamInfo,
+  size,
+}: UpstreamInfoIconProps) => {
   const intl = useIntl();
-  if (!upstreamInfo?.upstreamRef) {
+
+  if (!upstreamInfo) {
     return null;
   }
 
-  const iconProps = !upstreamInfo?.errorMessage
-    ? {
-      title: intl.formatMessage(messages.upstreamLinkOk),
-      ariaLabel: intl.formatMessage(messages.upstreamLinkOk),
-      src: Newsstand,
-    }
-    : {
+  let hasTwoIcons = false;
+  let secondIconProps = {};
```

Comment:
Nit: personally, instead of two separate variables, I would just put

```tsx
let secondIcon: JSX.Element | undefined;
...
    secondIcon = (
      <Icon
        size={size}
        title={intl.formatMessage(messages.upstreamLinkError)}
        ariaLabel={intl.formatMessage(messages.upstreamLinkError)}
        src={LinkOff}
      />
    );
```

```diff
-        {hasTwoIcons && (
-          <Icon
-            size={size}
-            {...secondIconProps}
-          />
-        )}
+        {secondIcon}
```